mvcc/watcher_group.go

```diff
@@ -219,7 +214,23 @@ func (wg *watcherGroup) delete(wa *watcher) bool {
 	return true
 }
 
-func (wg *watcherGroup) scanMinRev(curRev int64, compactRev int64) int64 {
+// choose selects watchers from the watcher group to update
+func (wg *watcherGroup) choose(maxWatchers int, curRev, compactRev int64) (*watcherGroup, int64) {
```

Comment:
seems like we can make this smarter by grouping the ones needs a lot of scan (low start revisions) in the same batch. but this should be a todo.